src/Atmos/Model/thermo_states.jl

```diff
@@ -2,6 +2,21 @@
 
 export new_thermo_state, recover_thermo_state
 
+"""
+    density(atmos::AtmosModel, state::Vars, aux::Vars)
+
+In the incompressible state, `state.ρ` that is used to
+extract intrinsic values (i.e. `e=state.energy.ρe/state.ρ`)
+is time invariant (by eliminating the tendencies) while
+`ref_state.ρ` is used to construct a thermodynamic state.
+`density` will get the `ref_state.ρ` for thermodynamic state
+when the model is incompressible.
```

Comment:
Ok. I cannot parse this comment at all (notwithstanding that I should know where this is going). Please rewrite.